kAI-api/src/middleware/error-handler.ts

```
@@ -0,0 +1,114 @@
+    const errorResponse = err.toJSON();
+
+    // Add rate limit headers if applicable
+    if ('retryAfter' in err && err.retryAfter) {
+      res.set('Retry-After', String(err.retryAfter));
+    }
+
+    return res.status(err.statusCode).json(errorResponse);
+  }
+
+  // Handle validation errors from express-validator
+  if (err instanceof Array && err.length > 0 && 'param' in err[0]) {
```

Biome (1.9.4)

[error] Use Array.isArray() instead of instanceof Array.

instanceof Array returns false for array-like objects and arrays from other execution contexts.
Unsafe fix: Use Array.isArray() instead.


(lint/suspicious/useIsArray)